### PR TITLE
Add a load balancer sample.

### PR DESCRIPTION
**Pull request by yschimke**
Uses a delegating Call.Factory.

### CODE REVIEW

**Comment by ZahraDress** [comment]
#

**Comment by swankjesse** [comment]
Is this an admission of OkHttpClient not doing enough internally?

File: samples/guide/build.gradle.kts [hunk 1]
```diff
@@ -8,12 +8,18 @@ dependencies {
...
   implementation(projects.okhttpTestingSupport)
   implementation(projects.okhttpTls)
+  implementation(projects.okhttpCoroutines)
   implementation(libs.animalsniffer.annotations)
   implementation(libs.squareup.moshi)
   implementation(libs.squareup.okio.fakefilesystem)
   kapt(libs.squareup.moshi.compiler)
 }
 
+task("execute", JavaExec::class) {
```
Comment (yschimke) [inline samples/guide/build.gradle.kts:18]: I'll remove

**Comment by yschimke** [comment]
Maybe, ok, yes.

But also that https://github.com/square/okhttp/issues/4530 isn't implemented

And that other "weird" requirements like implementing something like Happy Eyeballs over multiple networks might be a thing. Or that you want to send different requests over different networks (Bluetooth for small quick operations, LTE for large downloads).

**Comment by yschimke** [comment]
@swankjesse What do you think about this sample? Seems like a pattern we should allow for even if we don't implement client side loadbalancing ourselves. This would be valid in apps that want to use multiple connections to backends.

**Comment by yschimke** [comment]
Or we close it off...

**Comment by yschimke** [comment]
Closing, this PR is probably appropriate for now.

**Pull request closed without merging**